internal/post-comments/post-comments.go

```diff
@@ -0,0 +1,68 @@
+package postcomments
+
+import (
+	"grab/internal/comment"
+	"grab/internal/post"
+)
+
+type PostWithCommentsResponse struct {
+	Posts []PostWithComments `json:"posts"`
+}
+
+type PostWithComments struct {
+	ID       int64     `json:"id"`
+	Title    string    `json:"string"`
+	Comments []comment.Comment `json:"comments,omitempty"`
+}
+
+type PostCommentsWorker interface {
+	GetPostWithComments() (PostWithCommentsResponse, error)
+	combinePostWithComments([]post.Post, []comment.Comment) []PostWithComments
```

Comment:
You don't need to define this `combinePostWithComments` as an interface method because its only purpose is used in `GetPostWithComments`.

Reply:
btw, it's very weird when exported interface has unexported method inside